R/ocjr-drug-court-SQ780.R

```diff
@@ -1,17 +1,522 @@
+#   ) |>
+#   select(id, case_type, date_filed, count_as_disposed, disposition_date,
+#          disposition, charge_desc, offense_category_desc, offense_type_desc,
+#          probability, uccs_code, party, uccs_drug_plus, poss, drug,
+#          uccs_property, larceny, receive_stolen, embezzle, misc, pawn,
+#          hospitality, forgery)
+
+# This should be equivalent to the above, minus the `sq780_statute = case_when`
+# and `filter(count_as_disposed == "")` bits, but is hopefully more readable
+clean_df <- data_clean |>
+  mutate(
+    # Is the charge a drug possession charge covered by SQ780? -----------------
+    sq780_drugs = if_else(
+      # UCCS code or regex indicating drug possession, and no disqualifying regex
+      (uccs_drug_plus | regex_drugs) & !disqualifying_regex, TRUE, FALSE),
+    # Is the charge a property crime covered by SQ780? -------------------------
+    sq780_props = if_else(
+      # UCCS code or regex indicating 780 property crime, and no disqualifying regex
+      (uccs_property | regex_props) & !disqualifying_regex, TRUE, FALSE),
+    # Classify by specific statute ---------------------------------------------
+    # Simple possession
+    sq780_statute = case_when(
+      sq780_drugs ~
+        "63 O.S. Section 2-402 -- Simple Drug Possession",
+      # Grand / petit larceny
+      larceny & !receive_stolen ~
+        "21 O.S. Section 1704, 1705 -- Grand/Petit Larceny",
+      # Buying / receiving stolen property
+      larceny & receive_stolen ~
+        "21 O.S. Section 1713 -- Buying/Receiving/Concealing Stolen Property",
+      # Domesticated fish / game -- using new regex since it's split up above
+      str_detect(count_as_disposed, "(?i)fish|(?i)domesticated game") ~
+        "21 O.S. Section 1719.1 -- Taking Domesticated Gish/Game",
+      # Taking crude oil / gas / related
+      str_detect(count_as_disposed, "(?i)oil|drilling|gas") ~
+        "21 O.S. Section 1722 -- Unlawfully Taking Oil/Gas",
+      # Larceny of merchandise in retail or wholesale establishment
+      larceny & str_detect(count_as_disposed, "(?i)meat|edible|retail|wholesale") ~
+        "21 O.S. Section 1731 -- Larceny of Merchandise from Retail or Wholesale",
+      # Embezzlement
+      embezzle ~
+        "21 O.S. Sections 1451 -- Embezzlement",
+      # Defrauding hotel / lodging
+      hospitality ~
+        "21 O.S. Sections 1503 -- Defrauding Hospitality/Lodging",
+      # Vehicle Embezzlement
+      embezzle & str_detect(count_as_disposed, "(?i)vehicle|motor") ~
+        "21 O.S. Sections 1521 -- Vehicle Embezzlement",
+      # Fraud
+      str_detect(count_as_disposed, "cheat|defraud|personat") ~
+        "21 O.S. Sections 1541.1, 1541.2, 1541.3 -- Fraud",
+      # Pawn Shop Fraud
+      pawn ~
+        "59 O.S. Section 1512 -- Defrauding Pawn Shop",
+      # Forgery
+      forgery ~
+        "21 O.S. Section 1579, Section 1621 -- Forgery",
+      # Default
+      TRUE ~ "Unknown / Other"
+    )
```

Comment:
This bit still needs work, I wanted to be able to classify each charge by which SQ780 statute it related to, but we need to keep tweaking the regex and testing it more.